src/vm/functions/assets.rs

```diff
@@ -103,19 +93,13 @@ pub fn stx_transfer_consolidated(env: &mut Environment, from: &PrincipalData, to
     // loading from's locked amount and height
     // TODO: this does not count the inner stacks block header load, but arguably,
     // this could be optimized away, so it shouldn't penalize the caller.
+    // todo(ludo): double check this
```

Comment:
There should be additional memory accounting for loading the PoX lock state.

Reply:
ha, do you mind unpacking you thoughts? thanks!

Reply:
The lock state isn't 0 bytes.  The sender should be paying to load whatever lock state is necessary to carry out the token transfer.  Ideally, the sender does _not_ pay for loading any receiver lock state (since before, the sender only had to increase the receiver's unlocked token balance).  But, the sender _always_ pays for loading its own lock state, and if it needs to re-consolidate its now-unlocked tokens, it _may_ pay for storing its new lock state as well as its new balance.

Reply:
Now that we've reached consensus on using a single MARF read/write and a 48-byte `STXBalance` struct, this code should call `add_memory()` to accurately reflect the size of the `STXBalance` instances loaded.  Right now, it's undercounting the memory use.

Reply:
Addressed with https://github.com/blockstack/stacks-blockchain/pull/1850/commits/0c76e29ed9431ef3950a5ccf558f0dc568a617fc